src/services/ThirdPartyService.ts

```
@@ -127,4 +133,42 @@ export class ThirdPartyService implements IThirdPartyService {
             throw new Error(error.response.data);
         }
     }
+
+    public async syncDetail(userDetail: UserDetailRequest): Promise<void> {
+        let endPoint: string;
+        if (userDetail.source === VendorEnum.GPIB) {
+            endPoint = this.config.get(ConfigSettings.GPIB_API_ENDPOINT);
+
+            const authentication = await axios.post(
+                `${endPoint}/user/authenticate`,
+                JSON.stringify({
+                    username: userDetail.email,
+                    password: userDetail.password
+                }),
+                {
+                    headers: {
+                        "Content-Type": "application/json"
+                    }
+                }
+            );
+            if (authentication.status === 200) {
+                const token = authentication.data.token;
+
+                await axios.put(
+                    `${endPoint}/accountInfoes`,
+                    JSON.stringify({
+                        firstName: userDetail.firstName,
+                        lastName: userDetail.lastName,
+                        yob: moment(userDetail.dob, "DD/MM/YYYY").year()
+                    }),
+                    {
+                        headers: {
+                            "Content-Type": "application/json",
+                            Authorization: `Bearer ${token}`
+                        }
+                    }
+                );
+            }
+        }
+    }
```

should we throw or return error in unexpected case (else case of the two ifs) ?